### DISCUSSION TITLE
How to Hide status bar in a cross-platform Android application

### DISCUSSION THREAD

**Discussion by Rushikesh3009** [AvaloniaUI/Avalonia#13505] [Q&A] [answered]
How do I hide the status bar in a cross-platform application for Android..?

**Comment by Coloryr** [comment]
https://developer.android.google.cn/training/system-ui/status?hl=zh-cn

You can change the Java code to C#, like this 

```
public class MainActivity : AvaloniaMainActivity<App>
{

    protected override void OnCreate(Bundle savedInstanceState)
    {
        Window.SetFlags(WindowManagerFlags.Fullscreen, WindowManagerFlags.Fullscreen);
```

**Reply by Rushikesh3009** [reply]
Window.SetFlags(WindowManagerFlags.Fullscreen, WindowManagerFlags.Fullscreen);

This code works well on mobile phones, but on tablets, it doesn't hide the status.

**Reply by Coloryr** [reply]
What does `status` mean, it's the `ActionBar`?

**Reply by Rushikesh3009** [reply]
No, I means status bar of android tablets.

![StatusBar](https://github.com/AvaloniaUI/Avalonia/assets/149040605/1062e821-f949-45c3-914d-15774560fa81)

**Reply by Coloryr** [reply]
`Fullscreen` no work? what?
may need change 

https://developer.android.google.cn/training/system-ui/immersive?hl=zh-cn

**Reply by Rushikesh3009** [reply]
The full screen code is working on Android phones, but it's not working on Android tabs

**Comment by maxkatz6** [comment, marked as answer]
While you can use native APIs, it should be easier with InsetsManager as it abstracts it on supported platforms: https://docs.avaloniaui.net/docs/next/concepts/services/insets-manager

**Reply by Rushikesh3009** [reply]
Thank you so much @maxkatz6 for solution.

**Reply by Rushikesh3009** [reply]
Solution Provided by @maxkatz6 and @emmauss for Hide status bar in a cross-platform Android application or full screen for cross-platform Android application is:

 ```
protected override void OnLoaded(RoutedEventArgs e)
 {
     base.OnLoaded(e);
     var insetsManager = TopLevel.GetTopLevel(this)?.InsetsManager;

     if (insetsManager != null)
     {
         insetsManager.DisplayEdgeToEdge = true;
         insetsManager.IsSystemBarVisible = false;
     }
 }
```
Add the about function in `mainview.axaml.cs` class.
Refere #13598
